python/pyspark/mllib/tests.py

```diff
@@ -76,7 +76,8 @@
     pass
 
 ser = PickleSerializer()
-sc = SparkContext('local[4]', "MLlib tests")
+conf = SparkConf().set("spark.driver.allowMultipleContexts", "true")
```

Comment:
Yeah, you shouldn't set this. The parallelism in `dev/run-tests` will actually launch separate JVMs, so that's not the cause of this problem. In general, you should _never_ set `spark.driver.allowMultipleContexts` (it was only added as an escape-hatch backwards-compatibility option for a feature that we never properly supported).

There must be some other problem in the tests, likely due to test cleanup or SparkContext teardown not being executed properly.